test/unit-tests/broker-launch.test.js

```diff
@@ -1,26 +1,61 @@
+const sinon = require('sinon');
+const cp = require('child_process');
+const { EventEmitter } = require('events');
 const expect = require('chai').expect;
-const io = require('socket.io-client');
 
-describe('Broker connection handling', () => {
-    let client, dashboardServer;
+const launchBroker = require('../../dashboard/index');
 
-    before((done) => {
-        dashboardServer = require('../../dashboard/server/index');
-        client = io('http://localhost:5001/');
-        client.on('connect', done);
+// mock event emitter to simulate ChildProcess object
+const mockChildProcess = new EventEmitter();
+
+// sinon fakes for behaviour verification
+let fake = sinon.fake();
+let spawnArgs;
+
+describe('Broker daemonized launching', () => {
+    before(() => {
+        // stub cp.spawn, to avoid actual daemon launch
+        sinon.stub(cp, 'spawn').callsFake((...args) => {
+            spawnArgs = args;
+
+            mockChildProcess['unref'] = fake;
+            mockChildProcess['disconnect'] = fake;
+            return mockChildProcess;
+        });
+
+        // stub process.exit to avoid exiting
+        sinon.stub(process, 'exit');
```

Comment:
Assert that this stub is actually called with correct arguments

Reply:
Done ✨